runtime/nvqir/NVQIR.cpp

```diff
@@ -198,13 +256,7 @@ Array *__quantum__rt__qubit_allocate_array_with_state_complex64(
         size, data, cudaq::simulation_precision::fp64);
     return vectorSizetToArray(qubitIdxs);
   }
-  // Input was complex<double> but we prefer complex<float>. Make a copy,
-  // truncating the values.
-  cudaq::info("copying {} complex values from double to float.", size);
-  auto convertData = std::make_unique<std::complex<float>[]>(size);
-  for (uint64_t i = 0; i < size; ++i)
-    convertData[i] = std::complex<float>{static_cast<float>(data[i].real()),
-                                         static_cast<float>(data[i].imag())};
+  auto convertData = convertToComplex<float>(data, size);
```

Comment:
but here `size` looks like it is the number of complex values in the list?

Reply:
size here is a number of qubits - I printed it out to confirm. And we are using it to allocate qubits above as well.